src/features/auth/lib/authMachine.ts

```diff
@@ -58,26 +68,37 @@ type Farm = {
   verificationUrl?: string;
 };
 
+interface Authentication {
+  token?: Token;
+  rawToken?: string;
+  web3?: {
```

Comment:
Do these still need to be optional?

I imagine the web3 `user` fields would all be set at once

Reply:
There are different combinations of state. 

A logged in user has token and web3.
There is a stage where a guest user has a wallet, but has logged in yet.